app/src/androidTest/java/org/mozilla/focus/activity/GrantStoragePermissionTest.java

```diff
@@ -139,9 +141,16 @@ public void downloadImageAndDelete_deleteSuccessfully() {
         final int displayHeight = displayMetrics.heightPixels;
         onView(withId(R.id.main_content)).check(matches(isDisplayed())).perform(AndroidTestUtils.clickXY(displayWidth / 2, displayHeight / 2, Tap.LONG));
 
+        // Initialize DownloadCompleteIdlingResource and register content observer
+        downloadCompleteIdlingResource = new DownloadCompleteIdlingResource(activityRule.getActivity());
+        downloadCompleteIdlingResource.registerDownloadCompleteObserver();
+
```

Comment:
What will happen if we move downloadCompleteIdlingResource initialization after below line? If not thing will happen, I'd like to move them together (so init and register sits side by side)

Reply:
We cannot. Since we need to register content observer first before we start to download the file(i.e. Click "Save image").